flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryConfiguration.java

```diff
@@ -44,7 +48,13 @@
 	private static volatile MetricRegistryConfiguration defaultConfiguration;
 
 	// regex pattern to split the defined reporters
-	private static final Pattern splitPattern = Pattern.compile("\\s*,\\s*");
+	private static final Pattern reporterListPattern = Pattern.compile("\\s*,\\s*");
+
+	// regex pattern to extract the name from reporter configuration keys, e.g. "rep" from "metrics.reporter.rep.class"
+	private static final Pattern reporterClassPattern = Pattern.compile(
+		Pattern.quote(ConfigConstants.METRICS_REPORTER_PREFIX) +
+		"([\\S&&[^.]]*)\\." +
```

Comment:
It would be helpful to document that we are intersecting regex character classes.

Reply:
of course, lacking documentation is the one gripe I always have when i find a regex somewhere, and now i did that myself :(